### PR TITLE
update doc for Leaky ReLU

### PR DESCRIPTION
**Pull request by nswamy**
update documentation for split and LeakyReLU

@mli, @madjam, @zackchase, @jiajiechen, @Roshrini, @indhub

Please review.

### CODE REVIEW

File: src/operator/leaky_relu.cc [hunk 1]
```diff
@@ -22,14 +22,16 @@ Operator *LeakyReLUProp::CreateOperator(Context ctx) const {
 DMLC_REGISTER_PARAMETER(LeakyReLUParam);
 
 MXNET_REGISTER_OP_PROPERTY(LeakyReLU, LeakyReLUProp)
-.describe(R"code(Leaky ReLu activation
+.describe(R"code(Rectified Linear Unit that allows a small gradient when the unit is not active.
```
Comment (zackchase) [inline src/operator/leaky_relu.cc]: Should only be capitalized if introducing an acronym. Language is weird, change to:

"Leaky rectified linear unit activation function. Has a slope of one when the input is positive and a small positive slope when the input is negative."

File: src/operator/slice_channel-inl.h [hunk 2]
```diff
@@ -32,13 +32,13 @@ struct SliceChannelParam : public dmlc::Parameter<SliceChannelParam> {
...
-    .describe("Number of outputs to be sliced.");
+    .describe("Number of splits. Note that this should evenly divide the length of the `axis`.");
     DMLC_DECLARE_FIELD(axis).set_default(1)
-    .describe("Dimension along which to slice.");
+    .describe("Axis along which to split.");
     DMLC_DECLARE_FIELD(squeeze_axis).set_default(0)
-    .describe("If true, the dimension will be squeezed."
-              " Also, input.shape[axis] must be the same as `num_outputs`"
-              " when squeeze_axis is turned on.");
+    .describe("If true, removes the dimension `axis` from the output shape"
```
Comment (zackchase) [inline src/operator/slice_channel-inl.h]: What do you mean by "removes the dimension `axis`" ? this is confusing. You are confusing axes with dimensions.

File: src/operator/slice_channel.cc [hunk 3]
```diff
@@ -27,35 +27,47 @@ Operator* SliceChannelProp::CreateOperatorEx(Context ctx,
...
+  z = [[ 1.  2.]
+       [ 3.  4.]],
+      [[ 5.  6.]
+       [ 7.  8.]]
+
+  z[0].shape = (2, 2)
 
-When setting optional argument ``squeeze_axis=1``, then the *k*-dimension will
-be removed from the shape if it becomes 1::
+`squeeze_axis=1` removes the dimension `axis` from the output shape
```
Comment (zackchase) [inline src/operator/slice_channel.cc]: again "the dimension `axis`" is confusing.

File: src/operator/slice_channel-inl.h (hunk 2 above)
Comment (nswamy) [inline src/operator/slice_channel-inl.h]: it removes the kth dimension. axis = k? how can i describe this?

**Comment by zackchase** [comment]
Hi @nswamy. What's the status on these changes? @piiswrong @madjam can we merge this once the requested changes are made?

File: src/operator/slice_channel-inl.h (hunk 2 above)
Comment (zackchase) [inline src/operator/slice_channel-inl.h]: Hi Naveen did we resolve whether this function does the same thing as numpy squeeze? If so can we please use their language for this which I think is clear?

File: src/operator/leaky_relu.cc (hunk 1 above)
Comment (nswamy) [inline src/operator/leaky_relu.cc]: changed to 
"Applies Leaky rectified linear unit activation element-wise to the input.

Leaky ReLUs attempt to fix the "dying ReLU" problem by allowing a small `slope` 
when the input is negative and has a slope of one when input is positive."

File: src/operator/slice_channel-inl.h (hunk 2 above)
Comment (nswamy) [inline src/operator/slice_channel-inl.h]: This is not exactly as the numpy.squeeze. I changed it to "removes the dimension of length 1 from the output shape."

File: src/operator/slice_channel.cc (hunk 3 above)
Comment (nswamy) [inline src/operator/slice_channel.cc]: changed to "removes the dimension of length 1 from the output shape."

File: src/operator/slice_channel.cc [hunk 4]
```diff
@@ -29,33 +29,44 @@ DMLC_REGISTER_PARAMETER(SliceChannelParam);
...
 .describe(R"code(Splits an array along a particular axis into multiple sub-arrays.
 
-Assume the input array has shape ``(d_0, ..., d_n)`` and we slice it into *m*
-(``num_outputs=m``) subarrays along axis *k*, then we will obtain a list of *m*
-arrays with each of which has shape ``(d_0, ..., d_k/m, ..., d_n)``.
+.. note:: ``SliceChannel`` is depreacted. Use ``split`` instead.
 
-For example::
+**Note** that `num_outputs` should evenly divide the length of the axis along
+which the array will be split.
```
Comment (madjam) [inline src/operator/slice_channel.cc]: Suggest rewording the last part of the sentence as:

"... the axis along which to split the array."

Comment (nswamy) [inline src/operator/slice_channel.cc]: done.

File: src/operator/slice_channel.cc [hunk 5]
```diff
@@ -29,33 +29,45 @@ DMLC_REGISTER_PARAMETER(SliceChannelParam);
...
+   z = [[ 1.  2.]
+        [ 3.  4.]],
+       [[ 5.  6.]
+        [ 7.  8.]]
 
-  y = split(x, axis=0, num_outputs=4, squeeze_axis=1)
-  y[0] = [ 1.,  2.]  // (2,) vector
+   z[0].shape = (2, 2)
+
+`squeeze_axis=1` removes the dimension of length 1 along the `axis`
```
Comment (nswamy) [inline src/operator/slice_channel.cc]: squeeze_axis can be set to one only if the dimension reduces to length of 1.
@zackchase / @madjam can you suggest how to word this?

Comment (zackchase) [inline src/operator/slice_channel.cc]: Hi @nswamy - the axis has length one. not the dimension. 

perhaps "Removes the axis with length 1 from the array shape."
